packages/terra-popup/src/Popup.jsx

```diff
@@ -0,0 +1,370 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Portal from 'react-portal';
+import PopupContent from './_PopupContent';
+import PopupArrow from './_PopupArrow';
+import PopupOverlay from './_PopupOverlay';
+import TetherComponent from './_TetherComponent';
+import './Popup.scss';
+
+const MIRROR_LR = {
+  center: 'center',
+  left: 'right',
+  right: 'left',
+};
+
+const MIRROR_TB = {
+  middle: 'middle',
+  top: 'bottom',
+  bottom: 'top',
+};
+
+const BASE_WIDTH = 16;
+const BASE_HEIGHT = 9;
+
+const DIMENSIONS = [
+  '10x',
+  '25x',
+  '50x',
+  '75x',
+  '100x',
+];
+
+const DIMENSIONS_MAP = {
+  '10x': 10,
+  '25x': 25,
+  '50x': 50,
+  '75x': 75,
+  '100x': 100,
+};
+
+const COMBINE = () => {
+  const combinations = [];
+  for (let i = 0; i < DIMENSIONS.length; i += 1) {
+    for (let j = 0; j < DIMENSIONS.length; j += 1) {
+      combinations.push(`${DIMENSIONS[i]} ${DIMENSIONS[j]}`);
+    }
+  }
+  return combinations;
+};
+
+const DIMENSION_COMBINATIONS = COMBINE();
+
+const propTypes = {
+  /**
+   * Bounding container for the popup, will use window if no value provided.
+   */
+  boundingRef: PropTypes.func,
+  /**
+   * The children to be displayed as content within the popup.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * CSS classnames that are append to the arrow.
+   */
+  classNameArrow: PropTypes.string,
+  /**
+   * CSS classnames that are append to the popup content inner.
+   */
+  classNameContent: PropTypes.string,
+  /**
+   * CSS classnames that are append to the overlay.
+   */
+  classNameOverlay: PropTypes.string,
+  /**
+   * Attachment point for the popup, this will be mirrored to the target.
+   */
+  contentAttachment: PropTypes.oneOf(TetherComponent.attachmentPositions),
+  /**
+   * A string pair of breakpoint sizes, ('10x', '25x', '50x', '75x', '10x'), i.e. 'vertical horizontal'
```

Comment:
This comment seems a bit confusing, we might want to expand on the description to make it clearer for a new consumer.